examples/imitation_learning_openbot/policies.py

```diff
@@ -25,33 +25,33 @@ def __init__(self, config):
         assert self._config.IMITATION_LEARNING_OPENBOT.PID.PROPORTIONAL_GAIN_HEADING >= 0.0
         assert self._config.IMITATION_LEARNING_OPENBOT.PID.DERIVATIVE_GAIN_HEADING >= 0.0
         assert self._config.IMITATION_LEARNING_OPENBOT.PID.FORWARD_MIN_ANGLE >= 0.0
-        assert self._config.SIMULATION_CONTROLLER.SIMULATION_STEP_TIME_SECONDS > 0.0 
+        assert self._config.SIMULATION_CONTROLLER.SIMULATION_STEP_TIME > 0.0 
     
-    def reset(self, obs, env_step_info):
+    def reset(self, obs, trajectory):
         
-        self._trajectory = env_step_info["agent_step_info"]["trajectory_data"]
+        self._trajectory = trajectory[0]
         assert len(self._trajectory) >= 1
         self._index_waypoint = 1 # initialized to 1 as waypoint with index 0 refers to the agent initial position
-        self._position_xy_old = obs["state_data"][0:2]
-        self._yaw_old = obs["state_data"][4]
+        self._position_xy_old = obs["location"][0:2]
+        self._yaw_old = np.deg2rad(obs["rotation"][1])
 
     def step(self, obs):
-
+        
         # xy position of the next waypoint in world frame:
         position_xy_desired = self._trajectory[self._index_waypoint][0:2] # [x_des, y_des]
 
         # compute the relative agent-target pose from the raw observation dictionary
-        xy_position_error, yaw_error = get_relative_target_pose(position_xy_desired, obs["state_data"][0:2], obs["state_data"][4])
+        xy_position_error, yaw_error = get_relative_target_pose(position_xy_desired, obs["location"][0:2], np.deg2rad(obs["rotation"][1]))
```

Comment:
```
cm_to_m = 0.01

location_xy_current = obs["location"][0:2] * cm_to_m
location_xy_desired = self._path[self._waypoint_index][0:2]
location_xy_error   = np.linalg.norm(location_xy_desired - location_xy_current)

rotation_yaw_current = np.deg2rad(obs["rotation"][1])
heading_xy_current   = np.array([np.cos(rotation_yaw_current), np.sin(rotation_yaw_current)])
heading_xy_desired   = (location_xy_desired - location_xy_current) / np.linalg.norm(location_xy_desired - location_xy_current)
rotation_yaw_error   = np.arctan2(heading_xy_desired[1], heading_xy_desired[0]) - np.arctan2(heading_xy_current[1], heading_xy_current[0])

if rotation_yaw_error < -np.pi:
    rotation_yaw_error += 2*np.pi
if rotation_yaw_error > np.pi:
    rotation_yaw_error -= 2*np.pi

location_xy_prev  = self._location_xy_prev
rotation_yaw_prev = self._rotation_yaw_prev

location_xy_speed  = np.linalg.norm(location_xy_current - location_xy_prev) / self._config.SIMULATION_CONTROLLER.SIMULATION_STEP_TIME_SECONDS)
rotation_yaw_speed = np.linalg.norm(rotation_yaw_current - rotation_yaw_prev) / self._config.SIMULATION_CONTROLLER.SIMULATION_STEP_TIME_SECONDS)
```

Reply:
```
# multiply by -1.0 because otherwise a positive rotation error would cause us to turn right, but it should cause us to turn left
control_right = \
    -1.0 *
    ((rotation_yaw_error * self._config.IMITATION_LEARNING_OPENBOT.ROTATION_YAW_PROPORTIONAL_GAIN) + \
    (rotation_yaw_speed * self._config.IMITATION_LEARNING_OPENBOT.ROTATION_YAW_DERIVATIVE_GAIN))

# multiply by an exponential decay term to prioritize rotation errors, use LOCATION_XY_HALF_LIFE to control how agressively to prioritize rotation errors
control_forward = \
    np.pow(2.0, -abs(rotation_yaw_error) / self._config.IMITATION_LEARNING_OPENBOT.LOCATION_XY_HALF_LIFE) * \ 
    ((location_xy_error * self._config.IMITATION_LEARNING_OPENBOT.LOCATION_XY_PROPORTIONAL_GAIN) + \
    (location_xy_speed * self._config.IMITATION_LEARNING_OPENBOT.LOCATION_XY_DERIVATIVE_GAIN))

action = np.array([control_forward + control_right, control_forward - control_left])
action = np.clip(action, -self._config.IMITATION_LEARNING_OPENBOT.MAX_ACTION, self._config.IMITATION_LEARNING_OPENBOT.MAX_ACTION)

```

Reply:
Updated this